common/deliver/deliver.go

```diff
@@ -260,6 +264,16 @@ func (h *Handler) deliverBlocks(ctx context.Context, srv *Server, envelope *cb.E
 
 	logger.Debugf("[channel: %s] Received seekInfo (%p) %v from %s", chdr.ChannelId, seekInfo, seekInfo, addr)
 
+	// A goroutine is pinned to an os thread for file i/o.
+	// Throttling limits the number of os threads created when getting blocks from ledger.
+	if h.ThrottleSemaphore != nil {
+		if err := h.ThrottleSemaphore.Acquire(ctx); err != nil {
```

Comment:
I remember we had a no-op semaphore somewhere, didn't we? Can't we inject it instead and then we don't need to do this *if* statement here?

Reply:
Added a no-op semaphore.